File: Leetcode/2.两数相加.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *res = new ListNode(), *result = res;
        int count = 0;

        while (l1 || l2 || count) {
            int mid = count;
            if (l1) {
                mid += l1 -> val;
                l1 = l1 -> next;
            }
            if (l2) {
                mid += l2 -> val;
                l2 = l2 -> next;
            }
            res -> next = new ListNode(mid % 10);
            res = res -> next;
            count = mid / 10;
        }

        return result -> next;
    }
};

File: Leetcode/2.两数相加.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	var ans *ListNode = &ListNode{Val: 0}
	res := ans
	carry := 0
	for l1 != nil || l2 != nil {
		number := carry
		if l1 != nil {
			number += l1.Val
			l1 = l1.Next
		}
		if l2 != nil {
			number += l2.Val
			l2 = l2.Next
		}
		ans.Next = &ListNode{Val: number % 10}
		carry = number / 10
		ans = ans.Next
	}
	if carry > 0 {
		ans.Next = &ListNode{Val: carry}
	}
	return res.Next
}